### DISCUSSION TITLE
Is it a bug or my mistake?

### DISCUSSION THREAD

**Discussion by GF-Huang** [canton7/Stylet#230] [Q&A] [answered]
Why I get this error? But application runnable, shows 3 rows in `DataGrid,` seems works well.

```
System.Windows.Data Error: 40 : BindingExpression path error: 'NetworkDataItems' property not found on 'object' 
''RootViewModel' (HashCode=16654654)'. BindingExpression:Path=NetworkDataItems; DataItem='RootViewModel' 
(HashCode=16654654); target element is 'DataGrid' (Name=''); target property is 'ItemsSource' (type 'IEnumerable')
```

### RootViewModel.cs

![image](https://user-images.githubusercontent.com/4510984/110968781-e4155800-8392-11eb-81b6-528b18e6b043.png)

### RootView.xaml
![image](https://user-images.githubusercontent.com/4510984/110969204-600fa000-8393-11eb-9ca6-9683ffea7b5a.png)

### NetworkDataViewModel.cs
![image](https://user-images.githubusercontent.com/4510984/110969147-50905700-8393-11eb-8c67-19a9816cd089.png)

### NetworkDataView.xaml
![image](https://user-images.githubusercontent.com/4510984/110969076-3c4c5a00-8393-11eb-871e-d99402f02093.png)

**Comment by canton7** [comment, marked as answer]
In Stylet, [you use `s:View.Model` with `ContentControl`](https://github.com/canton7/Stylet/wiki/ViewModel-First):

```xaml
<ContentControl s:View.Model="{Binding NetworkDataViewModel}"/>
```

You're trying to use it with `local:NetworkDataView`, but it won't do anything in that case. So the DataContext of `NetworkDataView` is still set to `RootViewModel`, which is why you get your binding error.

**Reply by GF-Huang** [reply]
So I need to wrap `local:NetworkDataView` into `ContentControl` as its `Content`?

**Reply by canton7** [reply]
No. The entire point of `s:View.Model` is that it:

1. Takes the ViewModel you gave it (through the binding)
2. Finds the right View type
3. Instantiates an instance of that view
4. Sets up the view, by assigning its DataContext and ActionTarget
5. Sets the view as the ContentControl's content.

This is all explained in the wiki page I linked you to.

You do not manually specify the view type. Stylet works it out from the ViewModel's type. That is the point of the ViewModel-first approach.

**Reply by GF-Huang** [reply]
Works! Thank you!

**Reply by GF-Huang** [reply]
I found that it also works if I modify to this, because `TabItem` is a subclass of `ContentControl`:

![image](https://user-images.githubusercontent.com/4510984/110977820-93efc300-839d-11eb-8dd1-af43a021b79b.png)

Is there any potential risk lead to unexpected bugs?

**Reply by canton7** [reply]
That should be fine. See also https://github.com/canton7/Stylet/wiki/Screens-and-Conductors#conductortcollectiononeactive

**Comment by GF-Huang** [comment]
Hey @canton7 , I found that `<ContentControl s:View.Model="{Binding XxxViewModel}" />` does not supports design time display. Is this by design?

**Reply by canton7** [reply]
There's simply no way to make it work at design-time unfortunately.

**Reply by GF-Huang** [reply]
If provide an design time instance like `d:DataContext="{d:DesignInstance Type=XXX, IsDesignTimeCreatable=True}"`, will it works?

**Reply by canton7** [reply]
Try it
